packages/react/src/drafts/IssueLabel/getColorFromHex.ts

```diff
@@ -0,0 +1,138 @@
+ * @returns
+ */
+export const getColorsFromHex = (
+  colorHex: string,
+  colorScheme = 'light',
+  bgColor: string,
+): React.CSSProperties | undefined => {
+  if (typeof colorHex !== 'string' || colorHex.startsWith('#') === false) return undefined
+  // start values for light mode
+  let bgLightness = 96
+  let lightnessIncrement = -1
+  let ratio = 4.5
+  // start values for dark mode
+  if (colorScheme.startsWith('dark')) {
+    bgLightness = 16
+    lightnessIncrement = 1
+    ratio = 5.5
+  }
+
+  // get hue and saturation value from hex color
+  // eslint-disable-next-line prefer-const
+  let {h, s} = hexToHsluv(colorHex)
+
+  // avoid intense bright colors
+  if (h >= 58 && h <= 198 && s > 70) {
+    s = 70
+  }
+  /**
+   * creating a background color from the provided hex color
+   */
+  const {colorHex: backgroundColor, lightness: currentBgLightness} = getColorWithContrast(
+    hsluvToHex({h, s, l: bgLightness}),
+    bgColor,
+    1.2,
+    lightnessIncrement as 1 | -1,
+  )
+
+  // avoid intense bright colors
+  if (h >= 58 && h <= 316 && s > 80) {
+    s = 80
+  }
+  /**
+   * creating a text color from with a contrast ratio of at least 4.5 to the generated background color
+   */
+  const {colorHex: textColor} = getColorWithContrast(
+    hsluvToHex({h, s, l: 50}),
+    backgroundColor,
+    ratio,
+    lightnessIncrement as 1 | -1,
+  )
+
+  return {
+    '--label-bgColor-rest': backgroundColor,
+    '--label-bgColor-hover': hsluvToHex({h, s, l: currentBgLightness + 4 * lightnessIncrement}),
+    '--label-bgColor-active': hsluvToHex({h, s, l: currentBgLightness + 8 * lightnessIncrement}),
+    '--label-fgColor': textColor,
+    '--label-fgColor-hover': textColor,
+    '--label-fgColor-active': textColor,
+  } as React.CSSProperties
+  // backgroundColorHover: hsluvToHex({h, s, l: currentBgLightness + 4 * lightnessIncrement}),
+  //   backgroundColorPressed: hsluvToHex({h, s, l: currentBgLightness + 8 * lightnessIncrement}),
+}
+/**
+ * Changes the lightness of a hex color until the contrast ratio is reached and returns the new hex color
+ * @param colorHex — initial hex color
+ * @param bgHex — color to check the contrast against
+ * @param contrastRatio — the contrast ratio to reach
+ * @param increment — the direction to change the lightness depending on the color scheme
+ * @returns the new hex color
+ */
+const getColorWithContrast = (
+  colorHex: string,
+  bgHex: string,
+  contrastRatio: number,
+  increment: 1 | -1,
+): {colorHex: string; lightness: number} => {
+  // deconstruct color
+  const hsluv = hexToHsluv(colorHex)
+  let {l: lightness} = hsluv
+  const {h, s} = hsluv
+  // change lightness until contrast is reached
+  while (getContrast(colorHex, bgHex) < contrastRatio && lightness > 0 && lightness < 100) {
+    lightness += increment
+    colorHex = hsluvToHex({h, s, l: lightness})
+  }
+  // return hex color and ligthness
+  return {colorHex, lightness}
```

Comment:
Instead of moving between color formats and spending time converting between each, could we parse colors into HSL at the beginning, mutate the channels in those colors, and then output hex? Or would that not work?

Reply:
I think the issue is that `getContrast` does not work with `HSLuv`. We could look at if we can use a different color model witha library that offers a contrast check out of the box?

Reply:
I may be wrong, but it seems like all libraries do conversions under the hood, e.g. https://github.com/color-js/color.js/blob/3e97aa382b7a861409c8f1bdaf04fa7a42b1f5ee/src/contrast/WCAG21.js#L13 uses `getLuminance` which also does a conversion. 

What I am not sure about is: Is this lib going to be loaded at runtime? Because we need to run the code with the user input, right? If so, the diff between the size of the library may also be important.

Reply:
Thanks for clarifying, that totally makes sense about contrast checks needing to do a conversion no matter what since the WCAG luminance spec works off of rgb 👍